pkg/config/validators/validators.go

```diff
@@ -8,19 +8,52 @@ import (
 func GetValidator(variableKind string) func(string) error {
 	switch variableKind {
 	case "envVarMap":
-		return KeyValueMapValidator
+		return keyValueMapValidator
+	case "imagePullPolicy":
+		return imagePullPolicyValidator
+	case "kubernetesProbeType":
+		return kubernetesProbeTypeValidator
+	case "scalingResourceType":
+		return scalingResourceTypeValidator
 	default:
-		return DefaultValidator
+		return defaultValidator
 	}
 }
 
-func KeyValueMapValidator(input string) error {
+func imagePullPolicyValidator(input string) error {
+	switch input {
+	case "Always", "IfNotPresent", "Never":
+		return nil
+	default:
+		return fmt.Errorf("invalid image pull policy: %s", input)
```

Comment:
if this is returned to user we should include the valid options in the error?

Reply:
added allowed values within error response for applicable validators